src/common/buffer.cc

```diff
@@ -54,66 +56,65 @@ static std::atomic_flag buffer_debug_lock = ATOMIC_FLAG_INIT;
 # define bendl std::endl; }
 #endif
 
-  static atomic_t buffer_total_alloc;
-  static atomic64_t buffer_history_alloc_bytes;
-  static atomic64_t buffer_history_alloc_num;
+  static std::atomic<unsigned> buffer_total_alloc { 0 };
+  static std::atomic<int64_t> buffer_history_alloc_bytes { 0 };
```

Comment:
Thanks for the feedback! We have several places where there's signed/unsigned confusion-- including a couple of unsigned t signed casts that you'll see elsewhere that fixing would require interfaces changes to. Should I try to address those while we're in here?

I'll move atomic<int64_t> to atomic<uint64_t> for at least the low-hanging fruit in the meantime. (Meanwhile, just trying to get all these things to build...)